gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -1233,9 +1232,19 @@ public int nextInt() throws IOException {
   }
 
   /**
-   * Skips the next value recursively. If it is an object or array, all nested
-   * elements are skipped. This method is intended for use when the JSON token
-   * stream contains unrecognized or unhandled values.
+   * Skips the next value recursively. This method is intended for use when
+   * the JSON token stream contains unrecognized or unhandled values.
+   *
+   * <p>The behavior depends on the type of the next JSON token:
```

Comment:
I hope the wording here and below is fine; please let me know if you think it could be improved.